src/ios/VLCPlugin.m

```diff
@@ -325,48 +373,70 @@ - (void)playremotefile:(CDVInvokedUrlCommand*)command {
         pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"invalid remote file url"];
     }
     
-    [self vlc_sendPluginResult:pluginResult callbackId:command.callbackId];
+    [self vlc_sendPluginResult:pluginResult callbackId:self.callbackId];
 }
 
 - (void)pause:(CDVInvokedUrlCommand*)command {
+    [self pauseWithCallbackId:command.callbackId];
+}
+
+- (void)stop:(CDVInvokedUrlCommand*)command {
+    [self stopPlayback];
+}
+
+- (void)setaudioinfo:(CDVInvokedUrlCommand*)command{
+    NSDictionary  * info = [command.arguments  objectAtIndex:0];
+    [self vlc_setlockscreenmetadata:info refreshLockScreen:true];
+    
+    if ([self.audio isKindOfClass:[NYPRStream class]] && [info objectForKey:@"svg_logo"]) {
+        NYPRStream *stream = [NYPRStream createStream:[[NYPRPersistentStack sharedInstance] managedObjectContext]];
+        [stream loadFromDictionary:info];
+        self.audio = stream;
+        self.audio.state = self.mediaPlayer.state;
+        self.audioPlayerViewController.audio = self.audio;
+    }
+}
+
+- (void)vlc_setlockscreenmetadata:(NSDictionary*)metadata refreshLockScreen:(BOOL)refreshLockScreen {
+    self.lockScreenCache = [NSDictionary dictionaryWithDictionary:metadata];
+    if(refreshLockScreen){
+        [self vlc_setNowPlayingInfo:self.lockScreenCache retrieveLockscreenArt:YES];
+    }
+}
+
+#pragma mark Audio playback controls
+
+- (void)pauseWithCallbackId:(NSString *)callbackId {
     DDLogInfo (@"VLC Plugin pausing playback");
     if (self.mediaPlayer.isPlaying) {
         [self.mediaPlayer pause];
+        
+        if (self.audio) {
+            self.audio.state = NYPRAudioStatePaused;
+        }
     }
```

Comment:
We generally rely on state changes from the VLC player to track audio state (calling mediaPlayer.pause should trigger a state change to the paused state) -- is there a specific reason to set the state here to NYPRAudioStatePaused?